allennlp/training/trainer.py

```diff
@@ -323,11 +319,8 @@ class GradientDescentTrainer(Trainer):
         be useful to accommodate batches that are larger than the RAM size. Refer [Thomas Wolf's
         post][0] for details on Gradient Accumulation.
 
-    opt_level : `str`, optional, (default = `None`)
-        Each opt_level establishes a set of properties that govern Amp’s implementation of pure or mixed
-        precision training. Must be a choice of `"O0"`, `"O1"`, `"O2"`, or `"O3"`.
```

Comment:
I take it torch doesn't have these various optimization levels?  Do we lose anything substantial by not having that control?

Reply:
Right, torch doesn't have those. And I don't think we lose anything because there was really only one opt level worth using ("02").